src/VirtoCommerce.Platform.Caching/Redis/RedisPlatformMemoryCache.cs

```diff
@@ -1,47 +1,90 @@
 
 namespace VirtoCommerce.Platform.Redis
 {
     public class RedisPlatformMemoryCache : PlatformMemoryCache
     {
         private readonly ISubscriber _bus;
+        private readonly IMemoryCache _memoryCache;
+        private readonly CachingOptions _cachingOptions;
         private readonly RedisCachingOptions _redisCachingOptions;
         private readonly ILogger _log;
-        private readonly RetryPolicy _retryPolicy;
+        private readonly TelemetryClient _telemetryClient;
 
         private static readonly string _cacheId = Guid.NewGuid().ToString("N");
         
 
         public RedisPlatformMemoryCache(IMemoryCache memoryCache, IOptions<CachingOptions> options
+            , ConnectionMultiplexer connection
             , ISubscriber bus
+            , IOptions<CachingOptions> cachingOptions
             , IOptions<RedisCachingOptions> redisCachingOptions
             , ILogger<RedisPlatformMemoryCache> log
+            , TelemetryClient telemetryClient
             ) : base(memoryCache, options, log)
        {
+            _memoryCache = memoryCache;
             _log = log;
+            _telemetryClient = telemetryClient;
             _bus = bus;
 
+            _cachingOptions = cachingOptions.Value;
             _redisCachingOptions = redisCachingOptions.Value;
-            _bus.Unsubscribe(_redisCachingOptions.ChannelName);
-            _bus.Subscribe(_redisCachingOptions.ChannelName, OnMessage);
+
+            connection.ConnectionFailed += OnConnectionFailed;
+            connection.ConnectionRestored += OnConnectionRestored;
+
+            _bus.Subscribe(_redisCachingOptions.ChannelName, OnMessage, CommandFlags.FireAndForget);
 
             _log.LogInformation($"{nameof(RedisPlatformMemoryCache)}: subscribe to channel {_redisCachingOptions.ChannelName } current instance:{ _cacheId }");
+            _telemetryClient.TrackEvent("RedisSubscribed", new Dictionary<string, string>
+            {
+                {"channelName", _redisCachingOptions.ChannelName},
+                {"cacheId", _cacheId}
+            });
+        }
+
+        protected virtual void OnConnectionFailed(object? sender, ConnectionFailedEventArgs e)
+        {
+            _log.LogError($"Redis disconnected from instance { _cacheId }. Endpoint is {e.EndPoint}, failure type is {e.FailureType}");
+            _telemetryClient.TrackException(e.Exception);
+            _telemetryClient.TrackEvent("RedisDisconnected", new Dictionary<string, string>
+            {
+                {"channelName", _redisCachingOptions.ChannelName},
+                {"cacheId", _cacheId},
+                {"endpoint", e.EndPoint.ToString()},
+                {"failureType", e.FailureType.ToString()}
+            });
 
-            _retryPolicy = Policy.Handle<Exception>().WaitAndRetry(
-                _redisCachingOptions.BusRetryCount, retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt - 1)));
+            CacheEnabled = false;
```

Comment:
Please add some comments why you need to do this

Reply:
@tatarincev Done, review please